pytest-selenium/tests/test_search.py

```diff
@@ -0,0 +1,74 @@
+from random import choice
+from string import digits, ascii_letters
+
+from pages.content import Content
+from tests import markers
+
+
+@markers.test_case("C543235")
+@markers.parametrize("page_slug", ["preface"])
+@markers.nondestructive
+def test_message_when_search_yields_no_results(selenium, base_url, book_slug, page_slug):
+    # GIVEN: Book page is loaded
+    content = Content(selenium, base_url, book_slug=book_slug, page_slug=page_slug).open()
+    toolbar = content.toolbar
+    mobile = content.mobile
+    search_sidebar = content.sidebar.search_sidebar
+    page_url_before_search = content.current_url
+
+    # Using regex to create a random search term
+    search_term = "".join(choice(digits + ascii_letters) for i in range(25))
+
+    # WHEN: Search is triggered for a term which yields no results
+    if content.is_desktop:
+        toolbar.search_for(search_term)
+
+    if content.is_mobile:
+        mobile.search_for(search_term)
+
+    # THEN: Search sidebar displays no results message
+    assert (
+        search_sidebar.no_results_message == "Sorry, no results found for   ‘"
+        "{}"
+        "’".format(search_term)
+    )
+
+    # AND: User stays in the same page as before executing the search
+    assert content.current_url == page_url_before_search
+
+    # AND: Scroll position of the content is not changed during the search
+    if content.is_mobile:
+        toolbar.click_search_icon()
```

Comment:
why does this happen here? shouldn't it happen up on line 27? and why do you have to do it if `search_for` includes `trigger_search`?

Reply:
For mobile:
1. invoke `search_for` which clicks search icon -> enters search term -> hits enter on search textbox => all these happen in line 27
2. verify the 'sorry...' message in the search sidebar
3. verify the scroll position of content => since in mobile resolution search sidebar hides the content, I am closing the sidebar by clicking on the search icon (which is our toggle), in this step